### ISSUE TITLE
BackButtonIcon should never use iOS/MacOS style when running on Web

### ISSUE THREAD

**Issue by jacobsimionato** [flutter/flutter#107646]
BackButtonIcon selects a style depending on the underlying OS. This applies even when the app is Flutter Web e.g. running Chrome on Mac OS. I think that on kIsWeb, we should instead always use 'Icons.arrow_back' regardless of the underlying OS.

Current logic in [back_button.dart](https://github.com/flutter/flutter/blob/fb57da5f945d02ef4f98dfd9409a72b7cce74268/packages/flutter/lib/src/material/back_button.dart#L31):
```
  static IconData _getIconData(TargetPlatform platform) {
    switch (platform) {
      case TargetPlatform.android:
      case TargetPlatform.fuchsia:
      case TargetPlatform.linux:
      case TargetPlatform.windows:
        return Icons.arrow_back;
      case TargetPlatform.iOS:
      case TargetPlatform.macOS:
        return Icons.arrow_back_ios;
    }
  }
```

As a user, this feels weird because this style of back button, which is famously required everywhere in native iOS apps, is not expected by material app users on Web. Rationale:
- Users expect that Web app UX features that are a property of the site (for traditional DOM-based sites) are consistent across platforms. E.g. while you expect Chrome to have different scroll physics on MacOS vs Windows, you expect a site to have the same iconography across these platforms.
- Material seems to use the iOS-like arrows for a different specific purpose other than 'back' on Web - paging through content. See gmail and calendar screenshots below. Both sites use the traditional Material back arrow icon to represent 'back'.
- Apple itself doesn't even seem use the iOS back button to mean 'back' on Web, in fact there is basically no stack-based navigation on apple.com or iCloud.com . Apple does use the iOS-like back arrow on web for paging which is consistent with the material sites above.

See Gmail screenshot:
![Gmail screenshot](https://user-images.githubusercontent.com/6869388/178989726-5c1c967b-f158-4167-bd9b-9b616d602cea.png)
Calendar screenshot:
![Calendar screenshot](https://user-images.githubusercontent.com/6869388/178989741-0f1d4540-e9e0-4186-944e-4249ce1894ab.png)
Apple store screenshot:
![Apple back icon](https://user-images.githubusercontent.com/6869388/178991618-30e09f5f-8d31-4001-bedb-865bf2def74c.png)

**Comment by InMatrix** [comment]
Since `BackButton` is a widget in the Material library, following what [Material Web](https://github.com/material-components/material-web) does in web browsers seems to be the correct solution. 

A related issue, which goes beyond this widget, is that Flutter lacks a generic, minimally-styled widget set from which users can build custom widgets.

**Comment by jacobsimionato** [comment]
Thanks for taking a look @InMatrix ! In this case, I believe we should update the `_getIconData` logic to always use `Icons.arrow_back` for web (i.e. `kIsWeb`), as I believe Material Web library always uses this icon.

Is Flutter Framework eng team comfortable with this change?

**Comment by Caffeinix** [comment]
`PlatformAdaptiveIcons` has the same problem as `_getIconData`.  There is an `_isCupertino()` method that (incorrectly) returns `true` for `TargetPlatform.macOS` regardless of whether it is running in a web browser.  Among other things, this means that overflow menus created using `PopupMenuButton` will use the wrong icon by default on macOS web.

**Comment by HansMuller** [comment]
Will try to get this issue resolved as part of #118229
